spec/index.adoc

```diff
@@ -438,6 +438,13 @@ If the canister has a https://webassembly.github.io/spec/core/binary/modules.htm
 +
 It is recommended for the canister to have a custom section called "icp:public candid:service", which contains the UTF-8 encoding of https://github.com/dfinity/candid/blob/master/spec/Candid.md#core-grammar[the Candid interface] for the canister.
 
+* `/canister/<canister_id>/property/<name>` (blob):
++
+Users can set arbitrary canister properties via settings, see <<ic-create_canister>> and <<ic-update_settings>>. The (blob) is optional. There is a maximum combined size limit for all properties of 4KB. Properties may be used to enable or disable system features, e.g.:
```

Comment:
“Users“ is a bit confusing here :-). Also, this line isn’t about about setting them, merely about their location in the system tree. Maybe

```suggestion
The systems maintains for each canister a generic key-value store, called the “Canister properties”. The system does not interpret these keys and values, but other platform components (e.g. the HTTP gateway protocol or Internet Identity) can.

The system may impose a size limit on the properties; it allows at least 4KB of combined size of the keys keys and values. Values can be empty.

These properties are configured and kept independently from installing or uninstalling code.
```

Note that the value cannot be optional in the state tree, but we can explicit say that it can be empty.

Is it worth pointing out that these properties should ideally never be used to configure things that affect the execution of the canister code in a way that the original canister developer would have to think about? Else we are risking conflating the roles of the canister developer and the canister admin, which are of course both on the “other side“ of the interface, but worth keeping apart.

Reply:
I’m trying to think where we had strongly thought about this distinction before. Maybe it was uploading the candid interface? There we eventually put it into the canister module, which is of course the right thing.